Src/Base/AMReX_ParmParse.cpp

```diff
@@ -867,7 +867,7 @@ squeryarr (const ParmParse::Table& table,
     if ( num_val == 0 ) return true;
 
     int stop_ix = start_ix + num_val - 1;
-    if ( static_cast<int>(ref.size()) <= stop_ix )
+    if ( static_cast<int>(ptr.size()) != stop_ix + 1 )  // grow or shrink to fit
```

Comment:
I am not yet 100% happy with the fix, since it might resize a vector where the API might allow to only partially read values from the params....?

Not 100% sure about this API contract, as the changes on the passed vector are only partly documented in the header file.

If needed, we could also apply a resize fix one level up in `queryAdd` for `std::vector`...?